---
title: '2025-12 scanpy ScRNAseq Umap'
description: 'chart 01'
pubDate: 'Jan 02 2026'
heroImage: '../resource/chart01/head.png'
---
import ExpandableMD from '../../components/ExpandableMD.astro';
import ScnPlots from '../resource/chart01/Scn_plots.md';

<p style={{ marginTop: '0' }}>
  &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
  <span style={{ 
    fontSize: '1.8em', 
    textDecoration: 'underline',
    textDecorationThickness: '1.5px',
    textDecorationColor: '#5f5757ff',
    verticalAlign: 'baseline',
    display: 'inline-block',
    lineHeight: '1',
  }}>N</span>ow in this session, we will explore the analysis of single-cell RNA sequencing (scRNA-seq) data using Scanpy, a powerful Python library for single-cell genomics. We will focus on visualizing the data using UMAP (Uniform Manifold Approximation and Projection), a popular dimensionality reduction technique that helps to reveal the underlying structure of high-dimensional data.
</p>

<p>
  &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
  Here, the scanpy package is utilized to process and analyze single-cell RNA sequencing data. The UMAP algorithm is applied to reduce the dimensionality of the data, allowing for effective visualization of cell populations and their relationships in a two-dimensional space. This approach aids in identifying distinct cell types, states, and potential biological insights from complex datasets.
</p>

<p>
  &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
  After the clustering is defined, we can visualize the results using UMAP plots. These plots are basically scatter plots where each point represents a single cell, colored according to its cluster assignment or expression level of a specific gene. The color bars (cbar) alongside the plots indicate the gene expression levels of corresponding gene, providing a reference for interpreting the information in the plot.
</p>

> **Note**: the original scanpy package doesn't seem to have a good support for color bar(cbar) style customization. Therefore, we need to manually modify the color bar style in UMAP plots generated by scanpy to enhance the visual appeal and clarity of the data representation.

<p>
  &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
  To manually control the style of color bars, an important trick is creating the independent **axe** for them, so that we can fully control its position and size. Otherwise, the default color bar created by scanpy would be too dumb. The common approoch to control the color bar size is to pass the **fraction** and **shrink** parameters into the scanpy plotting function.
</p>

<p>
  &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
  However, these parameters can't achieve the desired effect, since the matplotlib (or blame the scanpy, i'm not sure) plotting logic will automatically adjust the layout of the figure (no matter you use **tight_layout()** function or not), causing the axe (or technically, the space belong to the color bars) to be compressed. By creating independent axes for the color bars, we can precisely define their dimensions and placement within the figure, ensuring that they are visually appealing and effectively convey the necessary information about the data.
</p>

> One practical example of doing this is using **fig.add_axes()** to manually design the color bar arrangement.

<p>
  &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
  This method provides greater flexibility and control over the appearance of the color bars. Notice that we need to carefully adjust the axes for the color bars, therefore enhance the interpretability of the visualizations, making it easier for viewers to understand the gene expression patterns represented in the data.
</p>

>By implementing these techniques, we can create more visually appealing UMAP plots that effectively communicate the insights derived from single-cell RNA sequencing data analysis using Scanpy.

<ExpandableMD title="click to view details">
  <ScnPlots />
</ExpandableMD>
